lib/SIL/IR/SILBuilder.cpp

```diff
@@ -638,17 +638,24 @@ DebugValueInst *SILBuilder::createDebugValue(SILLocation Loc, SILValue src,
                                              SILDebugVariable Var,
                                              PoisonRefs_t poisonRefs,
                                              UsesMoveableValueDebugInfo_t moved,
-                                             bool trace) {
+                                             bool trace, bool overrideLoc) {
   if (shouldDropVariable(Var, Loc))
     return nullptr;
 
   llvm::SmallString<4> Name;
 
-  // Debug location overrides cannot apply to debug value instructions.
-  DebugLocOverrideRAII LocOverride{*this, std::nullopt};
-  return insert(DebugValueInst::create(
-      getSILDebugLocation(Loc, true), src, getModule(),
-      *substituteAnonymousArgs(Name, Var, Loc), poisonRefs, moved, trace));
+  SILDebugLocation DebugLoc;
+  if (overrideLoc) {
+    // Debug location overrides cannot apply to debug value instructions.
+    DebugLocOverrideRAII LocOverride{*this, std::nullopt};
```

Comment:
Does putting this into the `{` block mean that it won't affect the new instruction? Is that what you want here?

Reply:
The loc override applied in the getSILDebugLocation below. I am keeping the old behavior consistent while allowing us to not override the loc in the else statement below while deserializing